src/contour/TerminalSession.cpp

```diff
@@ -819,15 +821,23 @@ void TerminalSession::executeAction(actions::Action const& _action)
 
 void TerminalSession::spawnNewTerminal(string const& _profileName)
 {
-    ::contour::spawnNewTerminal(
-        programPath_,
-        config_.backingFilePath.generic_string(),
-        _profileName,
-        [this]() -> string {
+    auto const wd = [this]() -> string {
             auto const _l = scoped_lock{terminal_};
             return terminal_.screen().currentWorkingDirectory();
-        }()
-    );
+        }();
+
+    std::cout << wd <<std::endl;
+    if (config_.spawnNewProcess) {
+        ::contour::spawnNewTerminal(
+            programPath_,
+            config_.backingFilePath.generic_string(),
+            _profileName,
+            wd
+        );
+    } else {
+        config_.profile(profileName_)->shell.workingDirectory = FileSystem::path(wd);
```

Comment:
This will change the WD also for the current instance. I'm not sure that's the best approach. What would work is that you create a copy of `config_`, then mutate and use that instead.

Reply:
Well in theory it won't change, because it's getting the working directory from that instance